### PR TITLE
SwarmNL milestone 2 revision

### PR DESCRIPTION
**Pull request by sacha-l**
#### Project Abstract

This is a revision to Milestone 2 for building SwarmNl, previously stipulated here: https://github.com/w3f/Grants-Program/pull/2201. 

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [x] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (Polkadot AssetHub (DOT, USDC & USDT) address in the application and bank details via email, if applicable).
- [x] I understand that an agreed upon percentage of each milestone will be paid in vested DOT, to the Polkadot address listed in the application.
- [x] I am aware that, in order to receive a grant, I (and the entity I represent) have to successfully complete a KYC/KYB check.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [x] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@sacha-lansky:matrix.org`

### CODE REVIEW

**Review by takahser — CHANGES_REQUESTED** [review]
@sacha-l I'd like to make a couple of points here.

- the price of M2 is now 3x more expensive than in the original application. Please note that due to this price increase, this Level 2 grant would become a Level 3 grant, hence you'd need 5 approvals from members of the grants committee rather than 3.
- On another note, I think the original weekly rate was $2k per FTE but now it's $2.4k, could you explain this price increase?
- Regarding the significantly expanded scope (now requiring 2.5x the original FTE), I believe @keeganquigley would be the most qualified to assess whether all these additional deliverables are necessary and if the time estimate is realistic, given his prior evaluation of M1.

File: applications/swarm-nl.md
Comment (semuelle) [inline applications/swarm-nl.md:193]:
```suggestion
This brings us to total estimation of 10 weeks with a 1.5 FTE. With hourly rates revised to 60 USD/hr at 40 hour per week, our estimations come to a 24000 USD (1 FTE) + 12000 (0.5 FTE). This brings the total to 36,000 USD for Milestone 2, to fund the work of a Lead Engineer + Protocol Researcher (Adedeji) and Project Lead + Developer (Sacha) required to ship the libraries' features for this milestone.
```

**Comment by sacha-l** [comment]
Thank you for taking the time to review @takahser.

The way we calculated things for Milestone 2 was indeed $2k per FTE, calculated at an hourly rate of $40 USD/hour at a 40 hour week. In this revision we have increased the hourly rate to $60 USD. 

Upon completing Milestone 1, I realized we had significantly miscalculated the projected allocation of resources and cost. Namely:

* I realized that the level of complexity and protocol development the work entails merits more than a $40 per hour salary. Just by looking at around at rates for similar work being anywhere between 80 and 120, feel the increase to 60 is a reasonable middle ground.
* The single FTE was calculated to include both our hours, i.e. 10 hours of Project Management, writing tests, writing docs etc. + 30 hours dedicated to development. Both roles turned out to be equally important, each taking an equal amount of time, which turned out to be a lot more than 40 hour weeks. In my role for example I spent a lot more time writing code. Deji spent a lot of time on docs and guides. We basically became one unit. In this revision for Milestone 2, we are trying to be more realistic on the amount of effort we realize will be needed to deliver to the best of our abilities which is why we now have 1.5 FTE (putting my commitment at 20 hours a week and Deji's at 40 hours).
* We realized the importance of research and testing and I hadn't anticipated this well. Our M1 delivery was projected on 6 weeks, but it actually took us 8. This is why in this revision we've allocated 2 weeks for research in M2.1; 2 weeks for review, testing and improvements in M2.2 + 2 more weeks for integration testing and benchmarking (which is tricky!) and 2 more weeks for testing and touching things up in the final deliverable, "Extendability of library". This is also why there's an increase in time

Overall, we adjusted the hourly rate to $60 and adjusted the projected workload to be more realistic based on the experience we had shipping M1. Hope this helps clarify.

**Comment by semuelle** [comment]
Hi @sacha-l, thanks for clarifying. Since your initial application, we have started paying out grants with a percentage in DOT (vested over two years). Would you be comfortable with switching to 50% DOT going forward?

**Comment by sacha-l** [comment]
Hi @semuelle - can you provide more details on what that means ? What are the advantages of receiving DOT vested over two years?

**Comment by semuelle** [comment]
It means, part of the payment is in DOT tokens and linearly unlocks over the span of two years. We have incorporated 50%+ vested DOT payments to incentivise grant recipients to participate in the ecosystem beyond the grant payout. See also https://wiki.polkadot.network/docs/learn-transactions#vested-transfers and https://grants.web3.foundation/docs/faq#what-does-it-mean-for-payments-to-be-at-least-50-dot.

In your case, if you choose 50% DOT, for M2, you would get 18,000 USD in fiat or stablecoin, and the remaining 18,000 USD as DOT, where each block on the network unlocks a small portion for two years.

**Comment by sacha-l** [comment]
Hi @semuelle - After carefully considering the locked DOT payment opportunity, we decided we would happily accept switching to this model going forward.

**Comment by thewoodfish** [comment]
> Hi @semuelle - After carefully considering the locked DOT payment opportunity, we decided we would happily accept switching to this model going forward.



> It means, part of the payment is in DOT tokens and linearly unlocks over the span of two years. We have incorporated 50%+ vested DOT payments to incentivise grant recipients to participate in the ecosystem beyond the grant payout. See also https://wiki.polkadot.network/docs/learn-transactions#vested-transfers and https://grants.web3.foundation/docs/faq#what-does-it-mean-for-payments-to-be-at-least-50-dot.
> 
> In your case, if you choose 50% DOT, for M2, you would get 18,000 USD in fiat or stablecoin, and the remaining 18,000 USD as DOT, where each block on the network unlocks a small portion for two years.

Thank you @semuelle , we think this is a great model that works. It will also help us further build and test our next project easily on the Polkadot network which we absolutely intend to do, in great style.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. Since @semuelle is currently OOO, let me quickly reply: In this case, could you add a relay chain address to the document? As far as I remember, you can only stake locked DOTs on the relay chain. Other than that, would you still be willing to reduce the price? Other committee members raised concerns about the significant increase.

**Comment by sacha-l** [comment]
Thanks for your reply @Noc2. We would be happy to discount our hours for this milestone push by 20%, which would bring the total cost to $28,800.00. With this we would request that we get the amount in USDC instead of locked DOT. Would that work on your end?

**Comment by thewoodfish** [comment]
Thank you @sacha-l for outlining what would works for us. We await your favourable response @Noc2, so we can work together to solve the problems and ship something great! 🚀

**Review by Noc2 — CHANGES_REQUESTED** [review]
Sorry for the delay here. At this stage, we need to follow the new rule: "At least 50% of any grant is paid out in vested DOT." and can not easily accept application that don't take vested DOTs.

**Comment by sacha-l** [comment]
> Sorry for the delay here. At this stage, we need to follow the new rule: "At least 50% of any grant is paid out in vested DOT." and can not easily accept application that don't take vested DOTs.

Understood. We would be happy to stick to the original plan and get more reviews from the committee in that case. If for some reason you advise against this, I would like to understand what the committee's reasoning might be. If this might be the case and because we just really want to get to work on shipping milestone 2, we would reduce some of the scope to keep it within the 30k tier.

**Comment by thewoodfish** [comment]
Thank you very much @sacha-l, we patiently await your feedback @Noc2. Thank you.

**Comment by keeganquigley** [comment]
Thanks @sacha-l just to confirm, you want to use the same Polkadot address for both USDC & DOT?

**Comment by sacha-l** [comment]
> Thanks @sacha-l just to confirm, you want to use the same Polkadot address for both USDC & DOT payments?

Yes we will use the same address for locked DOT payments and USDC. But we will use a different address than the last milestone payment. The address will be: 14pG7HieCjNk2hZ4X2YvxkbpAXs9oCeW8aNZcD7tD2tpaJT9

Has the committee come to a conclusion ?

**Comment by keeganquigley** [comment]
Thanks @sacha-l in that case, could you please update the Polkadot address at the top of the application as part of this commit? Since the address in the application has to match the one on the invoice. After that I'm willing to approve it myself, and then I can ping the other committee members again.

**Comment by keeganquigley** [comment]
Thanks @sacha-l can you also change the level from 2 to 3? Sorry I should have mentioned that also.

**Review by keeganquigley — APPROVED** [review]
Great, thanks @sacha-l I'm willing to approve, and will ping the rest of the team again to take a look.

**Review by semuelle — COMMENTED** [review]
Hi @sacha-l, I'm with David in that we should convert the remaining milestone to a 50% DOT payment. Alternatively, if the total remains under 30k I'd be willing to keep it as USDC/USDT payment, but that means the milestone would be capped at 18k USD.

**Comment by sacha-l** [comment]
Hi @semuelle thanks for you comment. To clarify, we would be happy to go forward with the original plan and accept locked DOT.

**Comment by semuelle** [comment]
> Hi @semuelle thanks for you comment. To clarify, we would be happy to go forward with the original plan and accept locked DOT.

Great to hear. In that case, could you please add a `DOT %: 50%` (or more) under the `Costs` line?

**Review by keeganquigley — DISMISSED** [review]
LGMT

**Comment by thewoodfish** [comment]
> > Hi @semuelle thanks for you comment. To clarify, we would be happy to go forward with the original plan and accept locked DOT.
> 
> Great to hear. In that case, could you please add a `DOT %: 50%` (or more) under the `Costs` line?

Hello @semuelle, we have made the changes requested. 
Thank you

**Review by takahser — CHANGES_REQUESTED** [review]
Any update on this?

File: applications/swarm-nl.md
Comment (sacha-l) [reply applications/swarm-nl.md:142]: M1 has been paid already. 36k is for M2. Does this clarify?

Comment (takahser) [reply applications/swarm-nl.md:142]: The application doc should reflect the costs of each milestone, as well as the total cost correctly. The fact that M1 has been paid already doesn't have an impact on this. Could you please rectify the amounts?

Comment (sacha-l) [reply applications/swarm-nl.md:142]: Ah understood now, updated in https://github.com/w3f/Grants-Program/pull/2363/commits/5b524d4f47af31f52e86e88ff03209b7f1743a55

**Comment by thewoodfish** [comment]
> @thewoodfish thx for the reminder. I just [commented](https://github.com/w3f/Grants-Program/pull/2363#discussion_r1791516860).

Always a pleasure ser! @sacha-l has made the change. I look forward to your response. Thank you!

**Review by keeganquigley — APPROVED** [review]

**Review by Noc2 — APPROVED** [review]

**Review by semuelle — APPROVED** [review]
LGTM

**Comment by thewoodfish** [comment]
Thank you very much @semuelle and @Noc2. We patiently await your approval @takahser. Thank you.

**Review by laboon — APPROVED** [review]

**Review by SBalaguer — APPROVED** [review]

**Pull request merged**
